### DISCUSSION TITLE
Module exports, how?

### DISCUSSION THREAD

**Discussion by twh270** [TriliumNext/Trilium#6155] [Q&A] [answered]
On JS-frontend, I've figured out that I can reference a function that's been exported in a child like this:

parent script: Utils.someFunc()
child script: exports.someFunc = async function ...

1. Is there a way to reference a function in a non-child note? 
2. Is there a way to reference a function without requiring the child note name as a prefix?

Second -- is it possible to export multiple functions from a backend script? I see in the tasks example that reconcileAssignments is module.exports = async function, which implies the type of module.exports is a function. I tried module.exports = { key: async function blah() } but that doesn't work.

**Comment by zadam** [comment, marked as answer]
Hello, 

> Is there a way to reference a function in a non-child note?

No, not possible ATM. I've been thinking about being able to define module dependencies using relation but it was just never necessary. Cloning seems to work quite well for this and you can nicely see the dependency tree.

> Is there a way to reference a function without requiring the child note name as a prefix?

No, you need to use the child note name. I think it's good practice to explicitly specify the module name.

Otherwise you can always do this:

```javascript
const {myFunc} = childModule;
```

And then you can call it just like `myFunc()` instead of `childModule.myFunc()`

> is it possible to export multiple functions from a backend script? I see in the tasks example that reconcileAssignments is module.exports = async function, which implies the type of module.exports is a function. I tried module.exports = { key: async function blah() } but that doesn't work.

Yes, that should work. Remember that it's JS so `module.exports` doesn't actually have any type and you can put anything there. I think your syntax was a bit wrong (function is declared as value in JS differently) and that's why it didn't work. This should work:

```javascript
module.exports = {
	myFunc: async () => {
	}
}
```

or:

```javascript
async function blah() {
	...
}

async function bleh() {
	...
}

module.exports = {
	blah,
	bleh
}
```
